docs/developer_guide.md

```diff
@@ -0,0 +1,59 @@
+# Setup Envoy AI Gateway Local Development Environment
+
+## Prerequisites
+- Go 1.23 or later
+- Envoy proxy installed
+- OpenAI API or AWS Bedrock key
+
+## Build Required Binaries
+
+```bash
+git clone https://github.com/envoyproxy/ai-gateway
+cd ai-gateway
+
+# Build the extproc binary
+go build -o out/extproc cmd/extproc/main.go
+```
+
+## Start Envoy
+
+Download Envoy from [release page](https://github.com/envoyproxy/envoy/releases). 
+
+The following instructions assume you have the `envoy` binary in your PATH.
+
+Use this [envoy.yaml](envoy.yaml) configuration, which is derived from the `test-extproc` test suite, as a starting point to set up the Envoy proxy.
+
+```bash
+envoy -c envoy.yaml
+```
+
+## Create External Processor Configuration
+
+Save the OpenAI API key to a file named `/tmp/openai-api-key`.
+
+Use this [extproc-config.yaml](extproc-config.yaml), which is also derived from the same test suite, to start ext_proc.
+
+
+```bash
+./out/extproc -configPath extproc-config.yaml
+```
```

Comment:
sorry this can be obsolete at any minute. could you simply just refer to `make help` command?